### PR TITLE
Feat/compounding 4626

### PR DESCRIPTION
**Pull request by robrobbins**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -206,7 +191,7 @@ compile_go_vaulttracker_test:
 
 compile_vaulttracker_test: compile_solidity_vaulttracker_test compile_go_vaulttracker_test
 
-compile_test: compile_adapters compile_fakes compile_tokens compile_swivel_test compile_marketplace_test compile_vaulttracker_test compile_mocks
+compile_test: compile_fakes compile_tokens compile_swivel_test compile_marketplace_test compile_vaulttracker_test compile_mocks
 
 # --------------------------------------------- CLEANING ----------------------------------------------------------------------------------------
```
Comment (robrobbins) [inline Makefile:197]: obviously adapters have been removed. this was some cruft remaining from before.

File: Makefile [hunk 2]
```diff
@@ -206,7 +191,7 @@ compile_go_vaulttracker_test:
...
```
Comment (robrobbins) [inline Makefile:196]: removing some cruft leftover from when adapters were present

File: test/mocks/Erc4626.sol [hunk 3]
```diff
@@ -22,6 +25,26 @@ contract Erc4626 {
...
+  function convertToAssets(uint256) public view returns (uint256) {
+    return convertToAssetsReturn;
+  }
+
+  // NOTE we refrain from a convertToAssetsCalled here as it's a hard coded 1e26
+  // also we would lose our ability to have it interfaced as `view`
+
+  function convertToAssetsReturns(uint256 a) public {
+    convertToAssetsReturn = a;
+  }
```
Comment (sourabhmarathe) [inline test/mocks/Erc4626.sol:46]: Please add natspec docs for these methods.

**Review by sourabhmarathe — APPROVED** [review]
Looks good, just needs natspec on one of the contracts with a bunch of functions added to it.

**Review by JTraversa — APPROVED** [review]
🚢

File: test/mocks/Erc4626.sol (hunk 3 above)
Comment (robrobbins) [reply test/mocks/Erc4626.sol:46]: this is a mock, but sure.

**Pull request merged**